frontend/src/components/GoalForm/index.js

```diff
@@ -188,19 +188,23 @@ export default function GoalForm({
       } catch (err) {
         setFetchError('There was an error loading your goal');
       } finally {
-        setIsLoading(false);
+        setIsAppLoading(false);
       }
     }
 
-    if (!fetchAttempted && !isNew && !isLoading) {
-      setIsLoading(true);
-    }
-
-    // only fetch once, on load, and only if the id isn't 'new'
-    if (!fetchAttempted && !isNew && isLoading) {
+    if (!fetchAttempted && !isNew && !isAppLoading) {
```

Comment:
@thewatermethod I merged these two statements as the 'isLoading' in the below IF statement was never true.